determined-ai/determined
# refactor: remove fluent bit & replace with slurm log shipper [DET-9704]

## Description
Remove fluent bit and deprecate agent configuration around it. Replaces fluent bit with the slurm log shipper, which will now require all task containers (including commands) to have python installed to report logs back to master.

If these configs are set by users, during agent startup a warning in logs is shown.
* Fluent.ContainerName
* Fluent.Image
* Fluent.Port

Task containers will require python to be installed to report logs back to master, if a custom image is used that do not have python installed the container will fail and informed it failed due to the missing python dependency.

Also removes redundant python checks in task entrypoints that call `task-logging-setup.sh`.




## Test Plan
- Spin up a devcluster w/ fluent configured on at least 1 agent
    - Confirm not fluent containers actually start & the following warning is displayed in logs: 
        - `fluent options have been set for the following agent: <AGENT_ID> support for fluent has been removed as of 0.24.1`
 - Run: `det cmd run --config environment.image=python "echo 'Hello World'"`
     - Confirm that "Hello World" is reported back in the CLI 
 - Run: `det cmd run --config environment.image=ubuntu "echo 'Hello World'"`
     - Confirm that the command fails with:
         - `container failed with non-zero exit code: required python dependency was not found in container`




## Commentary (optional)





## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.

## Ticket
DET-9704
